### PR TITLE
libstratosphere: Refactor everything.

### PR DESCRIPTION
**Pull request by SciresM**
Over the weekend, I re-wrote basically all of libstratosphere. It's much easier to use now! Also, the implementations for services are a lot closer to how N does it.

This'll merge soon, PR is just to give a little time for testing.

### CODE REVIEW

File: stratosphere/fs_mitm/fs_mitm.json [hunk 1]
```diff
@@ -58,6 +58,7 @@
 			"svcReplyAndReceiveWithUserBuffer":	"0x44",
 			"svcCreateEvent":	"0x45",
 			"svcCreateInterruptEvent":	"0x53",
+			"svcReadWriteRegister":	"0x4E",
```
[the remark this answers is not in the document]
Comment (SciresM) [reply stratosphere/fs_mitm/fs_mitm.json]: Haha, yes, it is.

**Comment by ThatNerdyPikachu** [comment]
> replaces eclct with set:sys mitm 

Genius move tbh, but why not a KIP?

**Comment by SciresM** [comment]
@ThatNerdyPikachu because it mitms settings, which is not a kip?

If something doesn't need to be a kip, it *should not be a kip*. That's the design philosophy I want to push.

**Comment by SciresM** [comment]
I have gotten really used to hitting "close and comment", wow.

**Comment by ThatNerdyPikachu** [comment]
@SciresM What happens when we run out of TIDs to replace?

**Comment by SciresM** [comment]
@ThatNerdyPikachu use fake tids. If they start logging that in telemetry, we'll deal with it then.

**Comment by bakugo** [comment]
The split set.mitm doesn't work in this since pm skips launching the module it replaces.

And this is just my opinion but I don't think replacing official modules with random custom modules that do completely different things is a very good idea, and this is an example of why. It's just going to cause confusion about what each module is and what it does. If a module does its own thing it would make more sense to give it its own titleid.

**Comment by SciresM** [comment]
@bakugo actually, it does work, because that's not how pm.boot2 works. My system displays the modified string on 6.1.0.

**Comment by bakugo** [comment]
It doesn't on mine (6.0.1), recompiled with [this comment](https://github.com/Atmosphere-NX/Atmosphere/blob/master/stratosphere/pm/source/pm_boot2.cpp#L102) removed and then it works.

Any idea why this might be happening?

**Comment by SciresM** [comment]
> It doesn't on mine (6.0.1), recompiled with [this comment](https://github.com/Atmosphere-NX/Atmosphere/blob/master/stratosphere/pm/source/pm_boot2.cpp#L102) removed and then it works.
> 
> Any idea why this might be happening?

Oh, yes, actually. One moment.

**Pull request merged**